File: packages/app/src/app/components/api-params.ts
export interface WebToFeedParams {
  url: string;
  linkPath: string;
  extendContent: string;
  contextPath: string;
  paginationPath: string;
  datePath: string;
  prerender: string;
  prerenderWaitUntil: string;
  filter: string;
  version: string;
  format: string;
  articleRecovery: string;
  prerenderScript: string;
  strictMode: string;
  eventFeed: string;
}

export const webToFeedParams: WebToFeedParams = {
  url: 'u',
  linkPath: 'l',
  extendContent: 'ec',
  contextPath: 'cp',
  paginationPath: 'pp',
  datePath: 'dp',
  prerender: 'p',
  prerenderWaitUntil: 'aw',
  filter: 'q',
  version: 'v',
  format: 'f',
  articleRecovery: 'ar',
  prerenderScript: 'ps',
  strictMode: 'sm',
  eventFeed: 'ef',
};

File: packages/server-core/src/main/kotlin/org/migor/feedless/api/ApiParams.kt
package org.migor.feedless.api

object ApiParams {
  const val corrId = "x-corr-id"
}

object WebToFeedParamsV2 {
  const val url = "u"
  const val linkPath = "l"
  const val extendContext = "ec"
  const val contextPath = "cp"
  const val paginationXPath = "pp"
  const val datePath = "dp"
  const val prerender = "p"
  const val prerenderWaitUntil = "aw"
  const val filter = "q"
  const val version = "v"
  const val format = "f"
  const val prerenderScript = "ps"
  const val strictMode = "sm"
  const val eventFeed = "ef"
}

object WebToFeedParamsV1 {
  const val url = "url"
  const val link = "l"
  const val extendContext = "x"
  const val contextPath = "context"
  const val datePath = "date"
  const val prerender = "p"
  const val prerenderWaitUntil = "aw"
  const val filter = "q"
  const val version = "v"
  const val format = "out"
  const val prerenderScript = "ps"
  const val debug = "debug"
  const val homepageUrl = "homepageUrl"
}


object WebToPageChangeParams {
  const val url = "u"
  const val prerender = "p"
  const val prerenderWaitUntil = "aw"
  const val version = "v"
  const val type = "t"
  const val format = "f"
  const val xpath = "x"
  const val prerenderScript = "ps"
}
